File: 6/valid_palindrome_2.py
class Solution:
    def validPalindrome(self, s: str) -> bool:
        start, end = 0, len(s) - 1
        while start < end:
            if s[start] == s[end]:
                start += 1
                end -= 1
            else:
                return s[start:end] == s[start:end][::-1] or s[start+1:end+1] == s[start+1:end+1][::-1]
        return True

File: 6/valid_palindrome_2.go
package main

import "fmt"

func isPalindrome(s []rune) bool {
	for i, j := 0, len(s)-1; i < j; i, j = i+1, j-1 {
		if s[i] != s[j] {
			return false
		}
	}
	return true
}

func validPalindrome(s string) bool {
	runes := []rune(s)
	for i, j := 0, len(s)-1; i < j; i, j = i+1, j-1 {
		if runes[i] == runes[j] {
			continue
		}
		return isPalindrome(runes[i:j]) || isPalindrome(runes[i+1:j+1])
	}
	return true
}

func main() {
	res := validPalindrome("eccer")
	fmt.Println(res)
}
